src/wp-includes/block-editor.php

```diff
@@ -243,6 +243,37 @@ function get_block_editor_settings( $editor_name, $custom_settings = array() ) {
 		$custom_settings
 	);
 
+	$editor_settings['__experimentalFeatures'] = WP_Theme_JSON_Resolver::get_merged_data( $editor_settings )->get_settings();
+
+	// These settings may need to be updated based on data coming from theme.json sources.
+	if ( isset( $editor_settings['__experimentalFeatures']['color']['palette'] ) ) {
```

Comment:
An alternative we've considered is to unset the legacy settings, but according to [this conversation](https://github.com/WordPress/gutenberg/issues/30082#issuecomment-816679803) we landed on retrofitting these values.

Reply:
https://github.com/WordPress/wordpress-develop/pull/1262#discussion_r634596983